_posts/2019-11-01-Crossaccount-Metrics-Forwarding.md

```diff
@@ -0,0 +1,420 @@
+    service_name = ''
+    cluster_name = ''
+    for dimension in metric['Dimensions']:
+        if dimension['Name'] == 'ClusterName':
+            cluster_name = dimension['Value']
+            break
+
+    for dimension in metric['Dimensions']:
+        if dimension['Name'] == 'ServiceName':
+            service_name = dimension['Value']
+            service_names.append(service_name)
+```
+
+This piece of code is used to derive name of cluster and services.
+
+Snippet 5 -
+```python
+metric_data_input = fetch_json_cloudwatch(
+    namespace='AWS/ECS',
+    dimensions=[
+        {
+            'Name': 'ClusterName',
+            'Value': cluster_name
+        },
+        {
+            'Name': 'ServiceName',
+            'Value': service_name
+        }
+    ],
+    cloudwatch_client=source_cloudwatch_client,
+    current_time=current_time
+)
+```
+
+This piece of code fetches metrics from source account for specified cluster and service name.  
+
+Snippet 6 -
+```python
+metric_data_output = {
+    'MetricName': metric_name,
+    'Dimensions': [
+        {
+            'Name': 'ClusterName',
+            'Value': cluster_name
+        },
+        {
+            'Name': 'ServiceName',
+            'Value': service_name
+        }
+    ],
+    'Values': [],
+    'StatisticValues': {
+        'SampleCount': 1,
+        'Sum': 0,
+        'Minimum': 0,
+        'Maximum': 0
+    },
+
+    'Timestamp': metric_data_input['MetricDataResults'][0]['Timestamps'][0],
+    'Unit': 'Percent'
+}
+```
+
+This piece of code create a dictionary in the format which can be used as input for the API(put_metric_data) which will be used to send metrics to destination account.
+
+Snippet 7 -
+```python
+resources = metric_data_input['MetricDataResults']
+
+for resource in resources:
+    label = resource['Label']
+    value = resource['Values']
+    # timestamp = resource['Timestamps'][0]
+    print(label)
+    label_components = label.split(' ')
+    metric = label_components[0]
+    stat = label_components[1]
+
+    if metric == 'MemoryUtilization':
+        if stat == 'Average':
+            metric_data_mem_output['Values'] = resource['Values']
+        else:
+            metric_data_mem_output['StatisticValues'][stat] = resource['Values'][0]
+
+    if metric == 'CPUUtilization':
+        if stat == 'Average':
+            metric_data_cpu_output['Values'] = resource['Values']
+        else:
+            metric_data_cpu_output['StatisticValues'][stat] = resource['Values'][0]
+
+metric_data_output = [
+metric_data_mem_output,
+metric_data_cpu_output
+]
+print(metric_data_output)
+```
+
+This piece of code retrieve name of metric and statistics from the fetched metrics. These names are used to filter fetched metrics so that metric value of CPUUtilization and MemoryUtilization for different statistics is populated to corresponding dictionaries. Finally, both the dictionaries are used to create a list which can be as input for the put_metric_data API.
```

Comment:
This piece of code retrieves **the name of the metric** and ... so that **the** metric value of  ...